tasks/main.yml

```diff
@@ -38,5 +38,45 @@
     dest: /etc/sysconfig/kdump
     backup: yes
     mode: "{{ __sysconfig_kdump.stat.mode | d('0644') }}"
-  when: kexec_crash_size.content|b64decode|int > 0
   notify: restart kdump
+
+- name: Find out reserved memory for the crash kernel
+  slurp:
+    src: /sys/kernel/kexec_crash_size
+  register: kexec_crash_size
+
+- name: Find out whether reserved memory for the crash kernel is loaded
+  slurp:
+    src: /sys/kernel/kexec_crash_loaded
+  register: kexec_crash_loaded
+
+- name: Set the kdump_reboot_required fact
+  set_fact:
+    kdump_reboot_required: "{{
+      (kexec_crash_size.content | b64decode | int < 1) or
+      (kexec_crash_loaded.content | b64decode | int != 1)
+    }}"
+
+- name: Fail if reboot is required and kdump_reboot_ok is false
+  fail:
+    msg: >-
+      "Reboot is required to apply changes. Re-execute the role after boot to
+      ensure that kdump is working."
+  when:
+    - kdump_reboot_required | bool
+    - not kdump_reboot_ok
+
+- name: Reboot the managed node
+  reboot:
+  when:
+    - kdump_reboot_required | bool
+    - kdump_reboot_ok | bool
```

Comment:
If the managed node was rebooted, you should reset `kdump_reboot_required` to `false` so that the user knows that the node was rebooted and does not need to be rebooted again - see https://github.com/linux-system-roles/kernel_settings/blob/master/handlers/main.yml

Reply:
Done. I do not think I need to pass the `| d(false)` filter because the role sets the default for the variables.